tests/functional/setups/manu_sneddon_2d.py

```diff
@@ -0,0 +1,425 @@
+"""Model setup for the fracturized pressure problem using Sneddon's analytical solution.
+
+References:
+
+    - [1] Starfield, C.: Boundary Element Methods in Solid Mechanics (1983)
+    - [2] Sneddon, I.N.: Fourier transforms (1951)
+
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Callable
+
+import numpy as np
+
+import porepy as pp
+from porepy.applications.convergence_analysis import ConvergenceAnalysis
+from porepy.applications.md_grids.model_geometries import (
+    SquareDomainOrthogonalFractures,
+)
+from porepy.geometry.distances import point_pointset
+
+
+def compute_eta(pointset_centers: np.ndarray, center: np.ndarray) -> np.ndarray:
+    """Compute the distance from fracture points to the fracture centre.
+
+    Parameter:
+        pointset_centers: Coordinates of fracture points.
+        center: Coordinates of the fracture center.
+
+    Return:
+        Array of distances each point in pointset to the center.
+
+    """
+    return point_pointset(pointset_centers, center)
+
+
+def get_bem_centers(
+    a: float, h: float, n: int, alpha: float, center: np.ndarray
+) -> np.ndarray:
+    """Compute coordinates of the centers of the BEM segments.
+
+    Parameters:
+        a: Half of the fracture length.
+        h: BEM segment length.
+        n: Number of BEM segments.
+        alpha: Orientation of the fracture.
+        center: Coordinates of the fracture center.
+
+    Return:
+        Array containing the BEM segment centers.
+
+    """
+
+    # See [1], p. 57 coordinate system (4.5.1)
+    bem_centers = np.zeros((3, n))
+    x_0 = center[0] - (a - 0.5 * h) * np.sin(alpha)
+    y_0 = center[1] - (a - 0.5 * h) * np.cos(alpha)
+    for i in range(0, n):
+        bem_centers[0, i] = x_0 + i * h * np.sin(alpha)
+        bem_centers[1, i] = y_0 + i * h * np.cos(alpha)
+
+    return bem_centers
+
+
+def analytical_displacements(
+    a: float, eta: np.ndarray, p0: float, G: float, poi: float
+) -> np.ndarray:
+    """Compute Sneddon's analytical solution for the pressurized fracture displacement.
+
+    Parameters:
+        a: Half of the fracture length.
+        eta: Distances of fracture points to the fracture center.
+        p0: Constant, fixed pressure.
+        G: Shear modulus.
+        poi: Poisson ratio.
+
+    Return
+        Array containing the analytical normal displacement jumps.
+
+    """
+    # See [2], p. 425 equ. 92
+    cons = (1 - poi) / G * p0 * a * 2
+    return cons * np.sqrt(1 - np.power(eta / a, 2))
+
+
+def transform(xc: np.ndarray, x: np.ndarray, alpha: float) -> np.ndarray:
```

Comment:
Only used once in below method, can be eliminated and code put there.
In general, all these helper methods before the classes are created seem to be part of the analytical solution and can be moved there. Would also help reduce the signatures.

Reply:
Good observation. Please implement!